cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -179,8 +179,7 @@ func runKubeletStartJoinPhase(c workflow.RunData) (returnErr error) {
 	// Now the kubelet will perform the TLS Bootstrap, transforming /etc/kubernetes/bootstrap-kubelet.conf to /etc/kubernetes/kubelet.conf
 	// Wait for the kubelet to create the /etc/kubernetes/kubelet.conf kubeconfig file. If this process
 	// times out, display a somewhat user-friendly message.
-	waiter := apiclient.NewKubeWaiter(nil, kubeadmconstants.TLSBootstrapTimeout, os.Stdout)
-	if err := waiter.WaitForKubeletAndFunc(waitForTLSBootstrappedClient); err != nil {
+	if err := waitForTLSBootstrappedClient(); err != nil {
```

Comment:
I'm not sure that getting read of WaitForKubeletAndFunc is the right thing to do.
As far as I understand the logic (that is really tangled) is:
- wait for waitForTLSBootstrappedClient to succeed 
or 
- wait for WaitForHealthyKubelet to fail

waitForTLSBootstrappedClient has a 2 minutes timeout
WaitForHealthyKubelet has a 40 sec sleep, and then starts polling with an exponential backoff

TL;DR: we wait for the TLSBootstrapped, but if after some times the kubelet is not healthy, we exit fast

Reply:
if we are keeping WaitForHealthyKubelet this has to be done in the following way:
- don't use WaitForKubeletAndFunc in this PR, potentially remove it because the API is weird. instead run checks serially.
- wait for 40 seconds for the kubelet to start; return on error
- wait for the kubelet.conf to be created - 5 minutes.

Reply:
I am ok with leaving the healthz check. It's just `WaitForKubeletAndFunc` and the short timeout of `waitForTLSBootstrappedClient` that are problematic ATM.
Certainly, even on a slow running system, the above call to `WaitForKubeletAndFunc` is going to exit after 40s. Hence, we need to ensure that `waitForTLSBootstrappedClient` is completed.

Reply:
> Hence, we need to ensure that waitForTLSBootstrappedClient is completed.

like i've mentioned above i'd prefer if we run these checks serially.
`WaitForKubeletAndFunc` is just a weird API.

Reply:
I got what @fabriziopandini is saying here and I think he's correct. Hence, I changed the PR to keep the `WaitForKubeletAndFunc` call, but also to have the new 5 minute timeout in `waitForTLSBootstrappedClient`. The polling interval there is also increased to 5 sec.

I do agree that `WaitForKubeletAndFunc` is a weird call. We can certainly replace it with something more appropriate, but let's not do it in this PR if we can.

Reply:
i think we should keep https://github.com/kubernetes/kubeadm/issues/2091 open, until we refactor WaitForKubeletAndFunc.

checking for kubelet.conf during kubelet /healthz errors, feels wrong.

Reply:
We probably need a different issue for `WaitForKubeletAndFunc`. After all, if we merge this PR, we would have fixed the original issue there.